src/aatoolbox/datasources/usgs/usgs_ndvi.py

```diff
@@ -0,0 +1,815 @@
+                stat=stat,
+                is_percentile=is_percentile,
+                kwargs=kwargs,
+            )
+
+        return self._processed_base_dir
+
+    def load(self, feature_col: str) -> pd.DataFrame:  # type: ignore
+        """
+        Load the processed USGS NDVI data.
+
+        Parameters
+        ----------
+        feature_col : str
+            String is  used as a suffix to the
+            processed file path for unique identication of
+            analyses done on different files and columns.
+            The same value must be passed to ``process()``.
+
+        Returns
+        -------
+        pd.DataFrame
+            The processed NDVI dataset.
+
+        Raises
+        ------
+        FileNotFoundError
+            If the requested file cannot be found.
+
+        Examples
+        --------
+        >>> from aatoolbox import create_country_config, \
+        ...  CodAB, UsgsNdviSmoothed
+        >>>
+        >>> # Retrieve admin 2 boundaries for Burkina Faso
+        >>> country_config = create_country_config(iso3="bfa")
+        >>> codab = CodAB(country_config=country_config)
+        >>> bfa_admin2 = codab.load(admin_level=2)
+        >>>
+        >>> # setup NDVI
+        >>> bfa_ndvi = UsgsNdviSmoothed(
+        ...     country_config=country_config,
+        ...     start_date=[2020, 1],
+        ...     end_date=[2020, 3]
+        ... )
+        >>> bfa_ndvi.download()
+        >>> bfa_ndvi.process(
+        ...    gdf=bfa_admin2,
+        ...    feature_col="ADM2_FR"
+        )
+        >>> bfa_ndvi.load(feature_col="ADM2_FR")
+        """
+        try:
+            processed_files = self._find_processed_files(
+                feature_col=feature_col
+            )
+            processed_dfs = [
+                self._load(filepath=fp, drop_modified=True)
+                for fp in processed_files
+            ]
+
+            df = functools.reduce(
+                lambda df1, df2: pd.merge(
+                    df1, df2, on=["date", "year", "dekad", feature_col]
+                ),
+                processed_dfs,
+            )
+        except TypeError as err:
+            raise FileNotFoundError(
+                "Files not found to load. Ensure the download() and process() "
+                "methods are called prior to load()."
+            ) from err
+
+        # filter loaded data frame between our instances dates
+        load_dates = expand_dekads(
+            dekad1=self._start_date, dekad2=self._end_date
+        )
+
+        loaded_dates = df[["year", "dekad"]].values.tolist()
+        keep_rows = [tuple(d) in load_dates for d in loaded_dates]
+
+        df = df.loc[keep_rows]
+
+        # put feature column as 1st column
+        df_feature_col = df.pop(feature_col)
+        df.insert(loc=0, column=feature_col, value=df_feature_col)
+
+        return df
+
+    def load_raster(
+        self, date: Union[date, str, Tuple[int, int]]
```

Comment:
Oops should have caught this earlier but you should rename the variable `date` since you also import `date` from `datetime`, to avoid confusion